systems/data-plan/base-rate/pkg/db/rate_repo.go

```diff
@@ -30,9 +30,9 @@ func (u *baseRateRepo) GetBaseRate(uuid uuid.UUID) (*Rate, error) {
 	return rate, nil
 }
 
-func (b *baseRateRepo) GetBaseRates(country, network, effectiveAt, simType string) ([]Rate, error) {
+func (b *baseRateRepo) GetBaseRates(country, network, effectiveAt string, simType SimType) ([]Rate, error) {
 	var rates []Rate
-	result := b.Db.GetGormDb().Where(&Rate{Country: country, Network: network, Sim_type: simType, Effective_at: effectiveAt}).Find(&rates)
+	result := b.Db.GetGormDb().Where(&Rate{Country: country, Network: network, SimType: SimType(simType), EffectiveAt: effectiveAt}).Find(&rates)
```

Comment:
Why casting `simType`  into `SimType`  when `simType` is already of type `SimType` ?

Reply:
addressed